src/components/Widgets/RelationControl.js

```diff
@@ -82,10 +94,33 @@ class RelationControl extends Component {
     return suggestion.data[valueField];
   };
 
+  getSuggestionDisplayValue = (suggestion) => {
+    const { field } = this.props;
+    const displayField = field.get('displayField');
+    return suggestion.data[displayField];
+  };
+
   renderSuggestion = (suggestion) => {
     const { field } = this.props;
-    const valueField = field.get('valueField');
-    return <span>{suggestion.data[valueField]}</span>;
+    // a suggestion should show its `displayField` value
+    const displayField = field.get('displayField');
+    return <span>{suggestion.data[displayField]}</span>;
+  };
+
+  renderInputComponent = (inputProps) => {
+    // extract the field `value`, `displayValue` and `id`
+    // separated from the other input props
+    // needed for the visible field to stay accessible
+    // https://github.com/moroshko/react-autosuggest#renderInputComponentProp
+    const { value, displayValue, id, ...otherInputProps } = inputProps;
+    // have the visible field display the `displayValue`
+    // and the hidden field store the real `value`
+    return (
+      <div>
+        <input {...otherInputProps} value={displayValue} />
+        <input type="hidden" id={id} value={value} />
```

Comment:
Still not understanding what this guy is doing. Can you clarify? Makes it seem like the `forID` prop value is getting real responsibility here, and I'm concerned it won't hold up, e.g. when nested.